server/src/main/java/org/elasticsearch/common/settings/KeyStoreWrapper.java

```diff
@@ -308,7 +308,9 @@ public void decrypt(char[] password) throws GeneralSecurityException, IOExceptio
         }
         if (formatVersion <= 2) {
             decryptLegacyEntries();
-            assert password.length == 0;
+            if (password.length != 0) {
+                throw new IllegalArgumentException("Keystore format does not accept non-empty passwords");
+            }
```

Comment:
2/2 non test change
This password is no longer internal API, there are code paths where the user supplies its value.